provider/asdf/install_plugin.go

```diff
@@ -0,0 +1,140 @@
+package asdf
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bitrise-io/bitrise/v2/log"
+	"github.com/bitrise-io/toolprovider/provider"
+)
+
+// Unlikely to conflict with any plugin name or URL, but clearly separates the plugin name and URL.
+const PluginSourceSeparator = "::"
+
+type PluginSource struct {
+	PluginName  string
+	GitCloneURL string
+}
+
+var pluginSourceMap = map[string]PluginSource{
+	"flutter": {PluginName: "flutter", GitCloneURL: "https://github.com/asdf-community/asdf-flutter.git"},
+	"tuist":   {PluginName: "tuist", GitCloneURL: "https://github.com/tuist/asdf-tuist.git"},
+	"ruby":    {PluginName: "ruby", GitCloneURL: "https://github.com/asdf-vm/asdf-ruby.git"},
+	"nodejs":  {PluginName: "nodejs", GitCloneURL: "https://github.com/asdf-vm/asdf-nodejs.git"},
+}
+
+func (a *AsdfToolProvider) InstallPlugin(tool provider.ToolRequest) error {
+	plugin, err := resolvePluginSource(tool)
+	if err != nil {
+		return fmt.Errorf("resolve plugin source for tool %s: %w", tool.ToolName, err)
+	}
+	if plugin == nil {
+		return nil
+	}
```

Comment:
What does this scenario mean? It's a bit hard to follow what the layers of functions do that leads to `plugin` being `nil`